### PR TITLE
[IOPID-702] : Feature flag for identity list

### PR DESCRIPTION
**Pull request by OraldoDoci**
## Short description

Feature flag to hide identity list until the API is created in future versions

## List of changes proposed in this pull request

- Flag in .env files to hide idp list

## How to test

1. Check in .env files that NEXT_PUBLIC_FEATURE_FLAG is set to false
2. Login
3. Check in pages /blocco-accesso and /blocco-accesso/operazione-completata the IDP list is hiden

### CODE REVIEW

File: src/app/[locale]/(pages)/blocco-accesso/operazione-completata/page.tsx [hunk 1]
```diff
@@ -61,7 +61,7 @@ const ProfileBlock = (): React.ReactElement => {
       />
       <CopyCodeCard code={addSpacesEvery3Chars(unlockCode)} />
 
-      {isIDPKnown && <IdpListOnApp />}
+      {isIDPKnown === 'true' && <IdpListOnApp />}
```
Comment (shadowsheep1) [inline src/app/[locale]/(pages)/blocco-accesso/operazione-completata/page.tsx:64]: I'll change `isIDPKnown` to `isIdpKnown`. I've seen many time the use of `Idp` so I suggesto to be consistent. What do you think?

File: src/app/[locale]/(pages)/blocco-accesso/page.tsx [hunk 2]
```diff
@@ -38,8 +38,8 @@ const ProfileBlock = (): React.ReactElement => {
       });
   };
 
-  const renderSummary = (isIDPKnown: boolean) => {
-    if (isIDPKnown) {
+  const renderSummary = (isIDPKnown?: string) => {
+    if (isIDPKnown === 'true') {
```
Comment (shadowsheep1) [inline src/app/[locale]/(pages)/blocco-accesso/page.tsx:42]: What about making an utility function to parse string 'true'/'false' present in our .env* files, and use it in the exported flags?

File: src/app/[locale]/(pages)/blocco-accesso/page.tsx [hunk 3]
```diff
@@ -59,7 +59,7 @@ const ProfileBlock = (): React.ReactElement => {
           summaryColumns={{ xs: 12, md: 7.5 }}
         />
         <Grid sx={{ maxWidth: '576px' }}>
-          {isIDPKnown && <IdpListOnApp />}
+          {isIDPKnown === 'true' && <IdpListOnApp />}
```
Comment (shadowsheep1) [inline src/app/[locale]/(pages)/blocco-accesso/page.tsx:62]: As above

File: src/app/[locale]/_utils/idps.ts [hunk 4]
```diff
@@ -85,4 +85,5 @@ const IDPS: { identityProviders: IdentityProvider[]; richiediSpid: string } = {
 
 export { IDPS };
 
-export const isIDPKnown = true;
+// TODO remove this temporary flag isIDPKnown after getSessionsList API is ready in a future version
+export const isIDPKnown = process.env.NEXT_PUBLIC_FEATURE_FLAG;
```
Comment (shadowsheep1) [inline src/app/[locale]/_utils/idps.ts:89]: Here we could make use of an utility function to convert 'true'/'false' value string in boolean values. What do you think about that?

**Review by shadowsheep1 — CHANGES_REQUESTED** [review]

File: src/app/[locale]/(pages)/blocco-accesso/operazione-completata/page.tsx (hunk 1 above)
Comment (OraldoDoci) [reply src/app/[locale]/(pages)/blocco-accesso/operazione-completata/page.tsx]: name changed to isIdpKnown

File: src/app/[locale]/_utils/idps.ts (hunk 4 above)
Comment (OraldoDoci) [reply src/app/[locale]/_utils/idps.ts]: made the utility function to convert env string value to boolean type value

**Review by shadowsheep1 — APPROVED** [review]
LGTM

**Pull request merged**